modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/AsyncStatus.java

```diff
@@ -37,4 +39,24 @@
      * Returns true if command is completed.
      */
     boolean isCompleted();
+
+    /**
+     * Gets the error message if any.
+     *
+     * @return the error message if any, {@code null} otherwise
+     * @since 11.5
+     */
+    default String getError() {
+        return null;
+    }
+
+    /**
+     * Checks if there is any error.
+     *
+     * @return {@code true} if there is any error, {@code false} otherwise
+     * @since 11.5
+     */
+    default boolean hasError() {
+        return StringUtils.isNotEmpty(getError());
```

Comment:
For a default method that gets overridden anyway in `BulkStatus` and is here only for the vanishingly unlikely case that someone has their own implementation of `AsyncStatus`, you should just do `return getError() != null`